src/pages/PaniniCreator.tsx

```diff
@@ -34,6 +37,40 @@ export default function PaniniCreator(props: PaniniCreatorProps) {
     console.log("reset");
   };
 
+  const randomizeOrderData = useCallback(() => {
+    function getRandomVal<T>(array: T[]): T {
+      return array[Math.floor(Math.random() * array.length)];
+    }
+    const formValues: SandwichPayload = methods.getValues();
+
+    Object.entries(PaniniNames).forEach(([keyName, stringPath]: [string, PaniniNames]) => {
```

Comment:
**issue:** This is super complicated, and very hard to maintain - you don't have to make it so generic in such dynamic way. You don't have any control over it now, and you have to struggle with types. Try to adjust your solution to this example:

```ts
export const randomPanini = (): SandwichFormValues => {
 function getRandomVal<T>(array: T[]): T {
      return array[Math.floor(Math.random() * array.length)];
    }

  return {
    sandwichName: generateRandomName(),
    cutlery: Math.random() < 0.5,
    napkins: Math.random() < 0.5,
    base: {
      bread: randomElementArray(breadVariants),
      cheese: [randomElementArray(cheeseVariants],
      ....
    },
    extras: {
      egg: [randomElementArray(eggVariants], 
      ....
      topping: Math.random() < 0.5 ? "SESAME" : null,
    },
  };
};
```